src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3592,14 +3592,14 @@ private BoundNode BindSimpleProgramCompilationUnit(CompilationUnitSyntax compila
                                                      expressionBody: null);
         }
 
-        private BoundNode BindRecordConstructorBody(RecordDeclarationSyntax recordDecl, BindingDiagnosticBag diagnostics)
+        private BoundNode BindPrimaryConstructorBody(TypeDeclarationSyntax typeDecl, BindingDiagnosticBag diagnostics)
         {
-            Debug.Assert(recordDecl.ParameterList is object);
-            Debug.Assert(recordDecl.IsKind(SyntaxKind.RecordDeclaration));
+            Debug.Assert(typeDecl.ParameterList is object);
+            Debug.Assert(typeDecl.Kind() is SyntaxKind.RecordDeclaration or SyntaxKind.ClassDeclaration);
```

Comment:
>) [](http://example.com/codeflow?start=103&length=1)

`or SyntaxKind.StructDeclaration`?

Reply:
No. Primary Constructors and record struct constructors do not have a body. Because a base initializer is the only thing that could be there, and they are not supported for structs